### PR TITLE
Azure data

### PR DESCRIPTION
**Pull request by BWC-TomW**
This PR adds data needed and logging policies for defender for endpoint and some additional Azure data sources.

this a resubmit of #14 to try and get the "Allow edits from maintainer setting" working.

### CODE REVIEW

**Comment by BWC-TomW** [comment]
@yugoslavskiy  - not entirely sure why but I don't seem to be able to create the PR with "allow edits set" - perhaps because the source repo is under an org.

**Comment by yugoslavskiy** [comment]
@BWC-TomW OK, let's try another approach. I will just comment on some sections that need adjustment (:

File: data_needed/DN_0109_defender_DeviceEvents.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
+title: DN_0109_defender_DeviceEvents
+author: Tom@BridewellConsulting
```
Comment (yugoslavskiy) [inline data_needed/DN_0109_defender_DeviceEvents.yml]: When there are special characters (`@`) in the field, we put quotation marks this way:

```
author: 'Tom@BridewellConsulting'
```

Could you please adjust it all over the documents?

File: data_needed/DN_0109_defender_DeviceEvents.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  Microsoft Defender for endpoint DeviceEvents table is a miscellaneous table used to contain information about events that do not have thier own table, for example: Defender Antivirus, exploit controls (ie ASR)
+loggingpolicy: 
+  - LP_0111_defender_for_endpoint
+references:
+  - https://docs.microsoft.com/en-us/microsoft-365/security/defender/advanced-hunting-deviceevents-table?view=o365-worldwide
+category: OS Logs                               # HTTP Logs | DNS Logs | IDS/IPS/NGFW Alerts | Antivirus Alerts | Network Flows | etc
+platform: Windows                   
+type: defender for endpoint                             
+channel: DeviceEvents                        
+
```
Comment (yugoslavskiy) [inline data_needed/DN_0109_defender_DeviceEvents.yml]: please remove empty line here

File: data_needed/DN_0112_defender_DeviceImageLoadEvents.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
+
```
Comment (yugoslavskiy) [inline data_needed/DN_0112_defender_DeviceImageLoadEvents.yml]: please remove empty line here

File: data_needed/DN_0113_defender_DeviceInfo.yml [hunk 4]
```diff
@@ -0,0 +1,40 @@
+
```
Comment (yugoslavskiy) [inline data_needed/DN_0113_defender_DeviceInfo.yml]: please remove empty line here

File: data_needed/DN_0115_defender_DeviceNetworkEvents.yml [hunk 5]
```diff
@@ -0,0 +1,54 @@
+
```
Comment (yugoslavskiy) [inline data_needed/DN_0115_defender_DeviceNetworkEvents.yml]: please remove empty line here

File: data_needed/DN_0116_defender_DeviceNetworkInfo.yml [hunk 6]
```diff
@@ -0,0 +1,31 @@
+
```
Comment (yugoslavskiy) [inline data_needed/DN_0116_defender_DeviceNetworkInfo.yml]: please remove empty line here

File: data_needed/DN_0125_azure_MMA_DnsEvents.yml [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+  - ResultCode 
+  - Severity 
+  - SourceSystem 
+  - SubType 
+  - TaskCategory 
+  - TimeGenerated
+  - Type 
+  - _ResourceId 
+  - _SubscriptionId 
+
```
Comment (yugoslavskiy) [inline data_needed/DN_0125_azure_MMA_DnsEvents.yml]: please remove empty line here

File: data_needed/DN_0137_azure_AD_AuditLogs.yml [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (yugoslavskiy) [inline data_needed/DN_0137_azure_AD_AuditLogs.yml]: please remove empty lines here

File: data_needed/DN_0138_azure_AD_AADNonInteractiveUserSignInLogs.yml [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (yugoslavskiy) [inline data_needed/DN_0138_azure_AD_AADNonInteractiveUserSignInLogs.yml]: please remove empty lines here

File: logging_policies/LP_0125_azure_MMA_DNS.yml [hunk 10]
```diff
@@ -0,0 +1,28 @@
+title: LP_0125_azure_MMA_DNS
+author: tom@Bridewell
+default: Not configured 
+volume: High             
+description: >
+ Azure DNS connector utilizies the (microsoft monitroing agent) to forward DNS logs and DNS zone information from a Windows DNS server to Azure Sentinel / OMS. this feauture is currently in public preview. 
+references:
+    - https://docs.microsoft.com/en-us/azure/sentinel/connect-dns
+configuration: |
+
```
Comment (yugoslavskiy) [inline logging_policies/LP_0125_azure_MMA_DNS.yml]: please remove empty line here

File: logging_policies/LP_0125_azure_MMA_DNS.yml [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+ Click Install agent on Azure Windows virtual machine.
+ In the Virtual machines list, select the DNS machine you want to stream into Azure Sentinel. Make sure this is a Windows VM.
+ In the window that opens for that VM, click Connect.
+ Click Enable in the DNS connector window.
+ If your DNS machine is not an Azure VM&#58; 
+
+ Click Install agent on non-Azure machines.
+ In the Direct agent window, select either Download Windows agent (64 bit) or Download Windows agent (32 bit).
+ Install the agent on your DNS machine. Copy the Workspace ID, Primary key, and Secondary key and use them when prompted during the installation.
+
```
Comment (yugoslavskiy) [inline logging_policies/LP_0125_azure_MMA_DNS.yml]: please remove empty line here

File: logging_policies/LP_0131_azure_OfficeActivity_SharePoint.yml [hunk 12]
```diff
@@ -0,0 +1,12 @@
+title: LP_0131_azure_OfficeActivity_SharePoint
+author: Tom@BridewellConsulting
+default: Partially (Other) # Not configured | Partially (Success) | Partially (Failure) | Partially (Other) | Configured
+volume: High             # Low | Medium | High | Extremely High
+description: >
+  Sharepoint and OneDrive auditing includes details such as who accessed files, file syncing actions and file sharing actions.   
+references:
+    - https://docs.microsoft.com/en-us/microsoft-365/compliance/search-the-audit-log-in-security-and-compliance?view=o365-worldwide#audited-activities
+configuration: |
+
```
Comment (yugoslavskiy) [inline logging_policies/LP_0131_azure_OfficeActivity_SharePoint.yml]: please remove empty line here

File: logging_policies/LP_0132_azure_OfficeActivity_Teams.yml [hunk 13]
```diff
@@ -0,0 +1,12 @@
+title: LP_0132_azure_OfficeActivity_Teams
+author: Tom@BridewellConsulting
+default: Partially (Other) # Not configured | Partially (Success) | Partially (Failure) | Partially (Other) | Configured
+volume: High             # Low | Medium | High | Extremely High
+description: >
+  Teams auditing logs certain actions related to teams managment and risky activities (ie sending URls). individual messages are not auditied.  
+references:
+    - https://docs.microsoft.com/en-us/microsoftteams/audit-log-events#teams-activities
+configuration: |
+
```
Comment (yugoslavskiy) [inline logging_policies/LP_0132_azure_OfficeActivity_Teams.yml]: please remove empty line here

File: logging_policies/LP_0133_azure_FrontDoor.yml [hunk 14]
```diff
@@ -0,0 +1,18 @@
+title: LP_0133_azure_FrontDoor
+author: Tom@BridewellConsulting
+default: Not configured # Not configured | Partially (Success) | Partially (Failure) | Partially (Other) | Configured
+volume: High             # Low | Medium | High | Extremely High
+description: >
+  Azure frontdoor diagnostic logging logs each HTTP request and WAF match to the AzureDiagnostics table.    
+references:
+    - https://docs.microsoft.com/en-us/microsoft-365/compliance/search-the-audit-log-in-security-and-compliance?view=o365-worldwide#audited-activities
+configuration: |
+
```
Comment (yugoslavskiy) [inline logging_policies/LP_0133_azure_FrontDoor.yml]: please remove empty line here

File: logging_policies/LP_0134_azure_AD_SigninLogs.yml [hunk 15]
```diff
@@ -0,0 +1,22 @@
...
+author: tom@Bridewell
+default: Not configured 
+volume: Medium             
+description: >
+   Azure AD SigninLogs contains information about interactive sign in attempts from users, it includes details about the login request, conditional access policies applied and the status. 
+references:
+    - https://docs.microsoft.com/en-us/azure/active-directory/reports-monitoring/howto-integrate-activity-logs-with-log-analytics
+    - https://docs.microsoft.com/en-us/azure/sentinel/connect-azure-active-directory
+configuration: |
+
```
Comment (yugoslavskiy) [inline logging_policies/LP_0134_azure_AD_SigninLogs.yml]: please remove empty line here

File: logging_policies/LP_0135_azure_AD_AuditLogs.yml [hunk 16]
```diff
@@ -0,0 +1,20 @@
...
+author: tom@Bridewell
+default: Not configured 
+volume: Low             
+description: >
+   The Azure AD AuditLogs contains logs about directory activities (for example group and user managment)
+references:
+    - https://docs.microsoft.com/en-us/azure/active-directory/reports-monitoring/howto-integrate-activity-logs-with-log-analytics
+    - https://docs.microsoft.com/en-us/azure/sentinel/connect-azure-active-directory
+configuration: |
+
```
Comment (yugoslavskiy) [inline logging_policies/LP_0135_azure_AD_AuditLogs.yml]: please remove empty line here

File: logging_policies/LP_0136_azure_AD_AADNonInteractiveUserSignInLogs.yml [hunk 17]
```diff
@@ -0,0 +1,20 @@
...
+author: tom@Bridewell
+default: Not configured 
+volume: Medium             
+description: >
+   The Azure AD AADNonInteractiveUserSignInLogs contains logs about non-interactive user logins. this includes logins across services using access tokens when the user is not required to reenter credentials. 
+references:
+    - https://docs.microsoft.com/en-us/azure/active-directory/reports-monitoring/howto-integrate-activity-logs-with-log-analytics
+    - https://docs.microsoft.com/en-us/azure/sentinel/connect-azure-active-directory
+configuration: |
+
```
Comment (yugoslavskiy) [inline logging_policies/LP_0136_azure_AD_AADNonInteractiveUserSignInLogs.yml]: please remove empty line here

File: logging_policies/LP_0137_azure_AD_AADServicePrincipalSignInLogs.yml [hunk 18]
```diff
@@ -0,0 +1,21 @@
...
+default: Not configured 
+volume: Medium             
+description: >
+    The Azure AD AADServicePrincipalSignInLogs contains logs about Service Principal logins. 
+    service principal sign-ins do not involve a user. Instead, they are sign-ins by any non-user account, such as apps or service principals (except managed identity sign-in) 
+references:
+    - https://docs.microsoft.com/en-us/azure/active-directory/reports-monitoring/howto-integrate-activity-logs-with-log-analytics
+    - https://docs.microsoft.com/en-us/azure/sentinel/connect-azure-active-directory
+configuration: |
+
```
Comment (yugoslavskiy) [inline logging_policies/LP_0137_azure_AD_AADServicePrincipalSignInLogs.yml]: please remove empty line here

File: logging_policies/LP_0138_azure_AD_AADManagedIdentitySignInLogs.yml [hunk 19]
```diff
@@ -0,0 +1,21 @@
...
+default: Not configured 
+volume: Medium             
+description: >
+    The Azure AD AADManagedIdentitySignInLogs contains logs about managed identity signins. 
+    managed identitys are azure resources that have screts managed by azure. 
+references:
+    - https://docs.microsoft.com/en-us/azure/active-directory/reports-monitoring/howto-integrate-activity-logs-with-log-analytics
+    - https://docs.microsoft.com/en-us/azure/sentinel/connect-azure-active-directory
+configuration: |
+
```
Comment (yugoslavskiy) [inline logging_policies/LP_0138_azure_AD_AADManagedIdentitySignInLogs.yml]: please remove empty line here

File: logging_policies/LP_0139_azure_AD_AADProvisioningLogs.yml [hunk 20]
```diff
@@ -0,0 +1,20 @@
...
+author: tom@Bridewell
+default: Not configured 
+volume: Low             
+description: >
+    The Azure AD AADProvisioningLogs contains logs Activities performed by the provisioning service, such as the creation of a group in ServiceNow or a user imported from Workday.
+references:
+    - https://docs.microsoft.com/en-us/azure/active-directory/reports-monitoring/howto-integrate-activity-logs-with-log-analytics
+    - https://docs.microsoft.com/en-us/azure/sentinel/connect-azure-active-directory
+configuration: |
+
```
Comment (yugoslavskiy) [inline logging_policies/LP_0139_azure_AD_AADProvisioningLogs.yml]: please remove empty line here

**Review by BWC-TomW — COMMENTED** [review]
Quotes added and empty lines removed

**Comment by yugoslavskiy** [comment]
Thank you very much, @BWC-TomW ! It's merged now (:

**Pull request merged**